cfn-resources/data-lake-pipeline/cmd/resource/resource.go

```diff
@@ -0,0 +1,353 @@
+	model.Profile = currentModel.Profile
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read Completed",
+		ResourceModel:   model}, nil
+}
+
+func ReadResponseModelGeneration(pe *admin.DataLakeIngestionPipeline) (model *Model) {
+	if pe != nil {
+		source := Source{
+			Type:           pe.Source.Type,
+			ClusterName:    pe.Source.ClusterName,
+			CollectionName: pe.Source.CollectionName,
+			DatabaseName:   pe.Source.DatabaseName,
+			GroupId:        pe.Source.GroupId,
+		}
+
+		partitionArr := []PartitionFields{}
+
+		for i := range pe.Sink.PartitionFields {
+			partitionField := PartitionFields{
+				FieldName: &pe.Sink.PartitionFields[i].FieldName,
+				Order:     &pe.Sink.PartitionFields[i].Order,
+			}
+			partitionArr = append(partitionArr, partitionField)
+		}
+		sink := Sink{
+			Type:             pe.Sink.Type,
+			MetadataProvider: pe.Sink.MetadataProvider,
+			MetadataRegion:   pe.Sink.MetadataRegion,
+			PartitionFields:  partitionArr,
+		}
+
+		transformationsArr := []Transformations{}
+		for i := range pe.Transformations {
+			transformations := Transformations{
+				Field: pe.Transformations[i].Field,
+				Type:  pe.Transformations[i].Type,
+			}
+			transformationsArr = append(transformationsArr, transformations)
+		}
+		createdStr := pe.CreatedDate.Format(time.RFC3339)
+		lastUpdatedStr := pe.LastUpdatedDate.Format(time.RFC3339)
+
+		models := Model{
+			ProjectId:       pe.GroupId,
+			Name:            pe.Name,
+			Id:              pe.Id,
+			CreatedDate:     &createdStr,
+			LastUpdatedDate: &lastUpdatedStr,
+			Sink:            &sink,
+			Source:          &source,
+			Transformations: transformationsArr,
+		}
+		return &models
+	}
+	return &Model{}
+}
+
+// Update handles the Update event from the Cloudformation service.
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(UpdateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlasClient, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	groupID := *currentModel.ProjectId
+	pipelineName := *currentModel.Name
+	dataLakeIntegrationPipeline := generateDataLakeIntegrationPipeline(currentModel)
+
+	updateRequest := atlasClient.AtlasV2.DataLakePipelinesApi.UpdatePipeline(ctx.Background(), groupID, pipelineName, dataLakeIntegrationPipeline)
+	pe, response, err := updateRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		if response.StatusCode == http.StatusBadRequest {
+			return handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          err.Error(),
+				HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, nil
```

Comment:
```suggestion
			return progress_events.GetFailedEventByCode(fmt.Sprintf("Error Updating resource: %s",err.Error()),
				cloudformation.HandlerErrorCodeAlreadyExists), nil
```

Reply:
here we are suggesting that everytime we get a badrequest is going to be an Already Exists, could be an input error